packages/esm-ohri-pmtct-app/src/post-submission-actions/mother-child-linkage-action.ts

```diff
@@ -53,11 +54,23 @@ async function constructPatientObjectFromObsData(
   obsGroup,
   encounterLocation: string,
   preferredIdentifierSource: string,
+  sessionMode: string,
 ): Promise<Patient> {
   // check if infant is alive
   const lifeStatusAtBirth = findChildObsInTree(obsGroup, infantLifeStatus);
+   // the infant is alive hence eligible for registration
   if (getObsValueCoded(lifeStatusAtBirth) == aliveStatus) {
-    // the infant is alive hence eligible for registration
+
+    const pTrackerId = findChildObsInTree(obsGroup, infantPTrackerId)?.value;
+    const existingpTrackerAssignee = await getIdentifierAssignee(pTrackerId, PtrackerIdentifierType);
+    if(existingpTrackerAssignee){
+      if(sessionMode === 'enter'){
+        throw new Error(`P Tracker Id (${pTrackerId}) already assigned to patient (${existingpTrackerAssignee})`);
+      }
+      else{
+        return null;
+      }
```

Comment:
This `else` looks redundant. Based on the logic, the error only throws in enter mode otherwise nothing should be done. Also doesn't this apply to edit mode as well?